backend/src/main/java/de/janka/capstonedrazen/controller/UserController.java

```diff
@@ -60,25 +57,46 @@ public UserController(UserService userService, PasswordService passwordService)
             @ApiResponse(code = SC_CONFLICT, message = "Unable to create User, user already exists")
     })
     public ResponseEntity<User> create(@AuthenticationPrincipal UserEntity authUser, @RequestBody User user) {
-        if (!authUser.getRole().equals("admin")) {
-            return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+        if (authUser.getRole().equals("admin")) {
+            try {
+                UserEntity userEntity = map(user);
+
+                UserEntity createdUserEntity = userService.create(userEntity);
+                User createdUser = map(createdUserEntity);
+                createdUser.setPassword(createdUserEntity.getPassword());
+                return ok(createdUser);
+
+            } catch (IllegalArgumentException e) {
+                return badRequest().build();
+            } catch (EntityExistsException e) {
+                return new ResponseEntity<>(HttpStatus.CONFLICT);
+            }
         }
+        return new ResponseEntity(HttpStatus.UNAUTHORIZED);
+
+    }
+
+    @PostMapping(value = "/createAsUser", produces = APPLICATION_JSON_VALUE, consumes = APPLICATION_JSON_VALUE)
+    @ApiResponses(value = {
+            @ApiResponse(code = SC_BAD_REQUEST, message = "Unable to create User with blank name"),
+            @ApiResponse(code = SC_CONFLICT, message = "Unable to create User, user already exists")
+    })
+    public ResponseEntity<User> createAsUser(@RequestBody User user) {
         try {
             UserEntity userEntity = map(user);
 
-            UserEntity createdUserEntity = userService.create(userEntity);
+            UserEntity createdUserEntity = userService.createAsUser(userEntity);
             User createdUser = map(createdUserEntity);
-            createdUser.setPassword(createdUserEntity.getPassword());
             return ok(createdUser);
 
         } catch (IllegalArgumentException e) {
```

Comment:
dasselbe wie oben, aber es ist so in Ordnung, nur vielleicht leserlicher mit gleich throwen statt try catch